docs/datagen/server/tags.md

```diff
@@ -0,0 +1,83 @@
+Tag Generation
+==============
+
+Tags can be generated for a mod by subclassing `TagsProvider` and implementing `#addTags`. After implementation, the provider must be [added][datagen] to the `DataGenerator`.
+
+`TagsProvider`
+--------------
+
+The tags provider has two methods used for generating tags: creating a tag with objects and other tags via `#tag`, or using tags from other object types to generate the tag data via `#getOrCreateRawBuilder`.
+
+!!! note
+    Typically, a provider will not call `#getOrCreateRawBuilder` directly unless a registry contains a representation of objects from a different registry (blocks have item representations to obtain the blocks in the inventory).
+
+When `#tag` is called, a `TagAppender` is created which acts as a chainable consumer of elements to add to the tag:
+
+Method           | Description
+:---:            | :---
+`add`            | Adds an object to a tag. 
+`addOptional`    | Adds an object to a tag through its name. If the object is not present, then the object will be skipped when loading.
+`addTag`         | Adds a tag to a tag. All elements within the inner tag are now a part of the outer tag.
```

Comment:
Do we need to clarify that this does not add all entries from the tag in the given tag to this tag, but only adds a reference to that tag?

Reply:
I don't think so. There's no way to check if a tag is within another tag, only whether the elements of a tag are in both tags. So, mentioning the encoded format of reference won't affect the users unless they are generating it manually, which goes against the point of referencing the datagen docs.